lib/galaxy/model/store/__init__.py

```diff
@@ -2215,6 +2217,14 @@ def add_dataset_collection(
     def add_dataset(self, dataset: model.DatasetInstance, include_files: bool = True) -> None:
         self.included_datasets[dataset] = (dataset, include_files)
 
+    def _ensure_dataset_file_exists(self, dataset: model.DatasetInstance) -> None:
+        state = dataset.dataset.state
+        if state in [model.Dataset.states.OK] and not dataset.file_name:
+            log.error(
+                f"Dataset [{dataset.id}] does not exists on on object store [{dataset.dataset.object_store_id or 'None'}], while trying to export."
+            )
+            raise Exception(f"Cannot export history dataset {dataset.hid}: {dataset.name} with id {self.exported_key(dataset.id)}")
```

Comment:
Arg, `hid` is only part of HistoryDatasetAssociation, but that should also be the only type of DatasetInstance you can have in a history ... I guess 
```suggestion
            raise Exception(f"Cannot export history dataset {getattr(dataset, 'hid', '')}: {dataset.name} with id {self.exported_key(dataset.id)}")
```
should fix the mypy error in a way that would also work for other items. Alternatively you could narrow the type with an `assert isinstance(dataset, HistoryDatasetAssociation)`

Reply:
I also fixed the exported_key parameter there.